tools/snp.sh

```diff
@@ -459,14 +442,41 @@ build_guest_initrd() {
 }
 
 save_binary_paths() {
-  local guest_kernel=$(ls $(realpath "${SETUP_WORKING_DIR}/AMDSEV/linux/guest/debian/linux-image/boot/vmlinuz*"))
-  local guest_kernel_version=$(ls "${guest_kernel}" | sed "s|.*/boot/vmlinuz-\(.*\)|\1|g")
-  GENERATED_INITRD_BIN="${SETUP_WORKING_DIR}/initrd.img-${guest_kernel_version}"
+  # Initialize variable based on linux distribution type
+  case ${ID,,} in
+    ubuntu)
+      local guest_kernel=$(ls $(realpath "${SETUP_WORKING_DIR}/AMDSEV/linux/guest/debian/linux-image/boot/vmlinuz*"))
+      local guest_kernel_version=$(ls "${guest_kernel}" | sed "s|.*/boot/vmlinuz-\(.*\)|\1|g")
+      
+      # Using initd for ubuntu based on /boot/
+      GENERATED_INITRD_BIN="${SETUP_WORKING_DIR}/initrd.img-${guest_kernel_version}"
+      ;;
+
+    rhel)
+      # Get snp guest version from snp guest kernel rpm pakage name
+      local guest_kernel_rpm_pckg=$(realpath ${SETUP_WORKING_DIR}/AMDSEV/linux/kernel-[0-9]*snp_guest*)
+      local rhel_kernel_name=$(basename "$guest_kernel_rpm_pckg")
+      local guest_kernel_version=$(echo "$rhel_kernel_name"| cut -d'-' -f2)
+
+      # Get Host CPU Architecture info( like: x86_64. x86.. so on)
+      # My Assumption: Guest CPU has same cpu architecture as host architecture 
+      # Reason: Guest using CPU type "host" may increase the VM performance
+      local host_arch=$(arch)
+      
+      # Copy and rename guest snp kernel from bzImage to vmlinuz
+      local bzImage_file=$(realpath ${SETUP_WORKING_DIR}/AMDSEV/linux/guest/arch/$host_arch/boot/bzImage)
+      cp -v $bzImage_file ${SETUP_WORKING_DIR}/AMDSEV/linux/guest/vmlinuz-$guest_kernel_version
+
+      # Using initramfs for RHEL based on /boot/
+      local guest_kernel=$(ls $(realpath "${SETUP_WORKING_DIR}/AMDSEV/linux/guest/vmlinuz*"))
+      GENERATED_INITRD_BIN="${SETUP_WORKING_DIR}/initramfs-${guest_kernel_version}.img"
+      ;;
+  esac
 
 # Save binary paths in source file
 cat > "${SETUP_WORKING_DIR}/source-bins" <<EOF
 QEMU_BIN="${SETUP_WORKING_DIR}/AMDSEV/qemu/build/qemu-system-x86_64"
-OVMF_BIN="${SETUP_WORKING_DIR}/AMDSEV/ovmf/Build/AmdSev/DEBUG_GCC5/FV/OVMF.fd"
+OVMF_BIN="${SETUP_WORKING_DIR}/AMDSEV/ovmf/Build/OvmfX64/DEBUG_GCC5/FV/OVMF.fd"
```

Comment:
Why was this changed? If you use the standard OVMF package, it does not have support for direct boot.

Reply:
I changed it as I could not get AmdSev package in ovmf folder.
I was using the package I got after AMDSEV build step from the snp-latest branch.